### PR TITLE
add dynamic updateFrequency via settingsRegistry

### PR DESCRIPTION
**Pull request by AjayThorve**
Implements #179. Also added the ability to set a custom refresh speed via jupyterlab settings. The settings once set will be global across the jupyterlab session, even when you restart the jupyterlab server. The setting also has type checking along with a minimum value of 1ms.

cc @jacobtomlinson 

Recording 2024-02-07 at 14 36 28

### CODE REVIEW

File: schema/config.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "title": "jupyterlab-nvdashboard",
+  "description": "Settings for the jupyterlab-nvdashboard extension.",
+  "type": "object",
+  "properties": {
+    "updateFrequency": {
+      "type": "integer",
+      "title": "Frequency of Updates",
+      "description": "The frequency of updates for the GPU Dashboard widgets, in milliseconds.",
+      "default": 1000,
```
Comment (jacobtomlinson) [inline schema/config.json:10]: I think we should change the default to something like `250`.

From using this in practice lately a lot of my code was running in less than a second and wouldn't register at all on the graphs.

**Review by jacobtomlinson — COMMENTED** [review]
I definitely appreciate making this configurable.

Have we done any testing to see what the performance impact is when setting it to smaller numbers?

The previous bokeh implementation updated many times per second with no noticeable impact. I think we should endeavour to set the poll rate as fast as we can without it taking noticeable amount of resources.

File: schema/config.json (hunk 1 above)
Comment (AjayThorve) [reply schema/config.json:10]: There is seemingly no performance hit on either 100 or 250, do we want to go as low as possible? In that case 100ms would should be possible.

**Comment by AjayThorve** [comment]
> I definitely appreciate making this configurable.
> 
> Have we done any testing to see what the performance impact is when setting it to smaller numbers?
> 
> The previous bokeh implementation updated many times per second with no noticeable impact. I think we should endeavour to set the default poll rate as fast as we can without it taking noticeable amount of resources.

@jacobtomlinson I have updated the default to 100ms, there is no performance impact at this value. Below 50ms is when I start to see some rendering backlog, but even then, you eventually see the entire timeline in a couple of seconds.

**Review by jacobtomlinson — APPROVED** [review]
This is great!

**Pull request merged**